Anni/wcicd (#16642)
cloudflare/cloudflare-docs: src/content/docs/workers/ci-cd/builds/index.mdx

---
pcx_content_type: concept
title: Builds
description: Use Workers Builds to integrate with Git and automatically build and deploy your Worker when you merge a pull request
sidebar:
  badge:
    text: Beta
---

Workers Builds allows you to connect an existing Worker to a GitHub or GitLab repository to set up automatic builds and deployments for your development workflow. 

## Get Started

To set up your Git integration with Workers:

1. **Select your Worker** in the dashboard and navigate to Settings > Build 
2. **Select the Git provider** you'd like to connect to by clicking "Connect". Follow the prompts to install the Cloudflare application on your git account.

> 3. **Configure your build settings** by selecting your desired git repository, branch, and configuring a deploy command and optionally a build command.

Reviewer: Seems like `build settings` should link to the configuration page in docs?

https://anni-wcicd.cloudflare-docs-7ou.pages.dev/workers/ci-cd/builds/build-configuration/

> 4. **Push a commit** to your repository to trigger your first build and deploy.

Reviewer: Nothing in Terraform? Not part of OpenAPI spec? Seems like a thing we should be explicit about somewhere in docs, not necessarily this page

## View Build Details

> Once a build is in progress, you can monitor its build logs and view build history by navigating to **"View build history"** on the deployments tab of your Worker. If successful, you can also view the build details for each new [version](https://developers.cloudflare.com/workers/configuration/versions-and-deployments/) created. ## Build Limits

Reviewer 2: "## Build Limits" snuck in here

---
pcx_content_type: concept

> title: Builds

Reviewer: <img width="331" alt="Screenshot 2024-09-18 at 2 08 49 PM" src="https://github.com/user-attachments/assets/bee8def2-b6ac-42d6-b99d-fa37ae8c53f7">

This is confusing, no?

Reviewer: This is still confusing. If we are calling the system, "Workers Builds" it is too generic to call this section just "Builds".

We need to be super clear here that the section currently called "Builds" is where to learn more about the built-in system that is native to the platform. I'm open to ideas about how we get there.

<img width="223" alt="Screenshot 2024-09-22 at 2 12 19 PM" src="https://github.com/user-attachments/assets/bd349b2b-81e3-4f20-84a0-b2624a133699">

Ex — this section is now super clear about this:

<img width="686" alt="Screenshot 2024-09-22 at 2 13 32 PM" src="https://github.com/user-attachments/assets/caa9fdf1-22e8-424c-98a0-00a60953484d">

How can we carry this through to the nav? I know tricky to get right but worth fighting for.

Reviewer: ```suggestion
title: Built-in CI/CD
```

...or something?

> description: Leverage Workers Builds to integrate with Git and automatically build and deploy your project when you make a change.

Reviewer: ```suggestion
description: Use Workers Builds to integrate with Git and automatically build and deploy your Worker when you merge a pull request
```

Specific?

sidebar:
  badge:
    text: Beta
---

Connect your existing Worker to a GitHub or GitLab repository to set up automatic build and deployment for your Workers development workflow.

## Get Started

To set up your Git integration with a Worker:

1. **Select your Worker** in the Dashboard and navigate to the Build settings of your Worker under the Settings tab.

> 2. **Select the Git provider** you’d like to connect to by clicking "Connect". Follow the prompts to install the Cloudflare application on your git account.

Reviewer 3: ```suggestion
2. **Select the Git provider** you’d like to connect to by selecting "Connect". Follow the prompts to install the Cloudflare application on your git account.
```

3. **Configure your build settings** by selecting your desired git repository, branch and configuring a deploy command.
4. **Push a commit** to your repository to trigger your first build and deploy.

If you have a monorepo or [Wrangler Enviornments](/workers/wrangler/environments/) setup, reference [Build Configuration](/workers/ci-cd/builds/build-configuration).

## Build Details

> When a build is in progress you can monitor the build by viewing its logs. Navigate to the "View build history" on the deployments page, and select "View Build". The build can take on the status of in progress, queued, successful, failed and canceled builds.

Reviewer 3: ```suggestion
When a build is in progress you can monitor the build by viewing its logs. Navigate to "View build history" on the deployments page, and select "View Build". The build can take on the status of in progress, queued, successful, failed and canceled builds.
```

---
pcx_content_type: concept
title: Builds
description: Leverage Workers Builds to integrate with Git and automatically build and deploy your project when you make a change.
sidebar:
  badge:
    text: Beta
---

Connect your existing Worker to a GitHub or GitLab repository to set up automatic build and deployment for your Worker.

## Get Started

To set up your Git integration with a Worker:

1. **Select your Worker** in the Dashboard and navigate to the Build settings of your Worker under the Settings tab.
2. **Select the Git provider** you’d like to connect to by clicking "Connect". Follow the prompts to install the Cloudflare application on your git account.
3. **Configure your build settings** by selecting your desired git repository, branch and configuring a deploy command.
   :::caution
   In Closed Beta, only GitHub respositories and branches named `Main` can be cloned successfully.
   :::
4. **Push a commit** to your repository to trigger your first build and deploy.

> If you have a monothelic repository or [Wrangler Enviornments](/workers/wrangler/environments/) setup, reference [Build Configuration](/workers/ci-cd/builds/build-configuration).

Reviewer 4: ```suggestion
If you have a monolithic repository or [Wrangler Enviornments](/workers/wrangler/environments/) setup, reference [Build Configuration](/workers/ci-cd/builds/build-configuration).
```
